### PR TITLE
toggling k8s configuration with a relevant property fixes gh-585

### PR DESCRIPTION
**Pull request by Haybu**

### CODE REVIEW

**Review by ryanjbaxter — COMMENTED** [review]
Should we make similar changes here?
https://github.com/spring-cloud/spring-cloud-kubernetes/blob/master/spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/reload/KubernetesClientConfigReloadAutoConfiguration.java

Checstyle is failing as well

**Comment by Haybu** [comment]
I considered changing it before but didn't see the original condition on class `@ConditionalOnProperty(value = "spring.cloud.kubernetes.enabled", matchIfMissing = true)` thought there is an implementation change that made the properties obsolete. If all the same, I can apply the same modified condition and tests.

**Comment by ryanjbaxter** [comment]
Sounds good

File: spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/main/resources/bootstrap-kubernetes.yaml [hunk 1]
```diff
@@ -2,7 +2,10 @@ spring:
   cloud:
     kubernetes:
       enabled: true
+      config:
+        enabled: true
```
Comment (ryanjbaxter) [inline spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/main/resources/bootstrap-kubernetes.yaml:6]: Isn't this the default?

File: spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/reload/KubernetesClientConfigReloadAutoConfiguration.java [hunk 2]
```diff
@@ -54,7 +54,7 @@
  * @author Ryan Baxter
  */
 @Configuration(proxyBeanMethods = false)
-@ConditionalOnKubernetesEnabled
+@Conditional(ConditionalKubernetesAndConfigEnabled.class)
```
Comment (ryanjbaxter) [inline spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/reload/KubernetesClientConfigReloadAutoConfiguration.java:57]: Shouldn't this be `@ConditionalKubernetesAndConfigEnabled`?

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]

File: spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/main/resources/bootstrap-kubernetes.yaml (hunk 1 above)
Comment (Haybu) [reply spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/main/resources/bootstrap-kubernetes.yaml]: correct. it should be the default.

File: spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigReloadAutoConfiguration.java [hunk 3]
```diff
@@ -60,7 +60,8 @@
  * @author Kris Iyer
  */
 @Configuration(proxyBeanMethods = false)
-@ConditionalOnProperty(value = "spring.cloud.kubernetes.enabled", matchIfMissing = true)
+@ConditionalOnProperty(value = { "spring.cloud.kubernetes.config.enabled", "spring.cloud.kubernetes.enabled" },
```
Comment (ryanjbaxter) [inline spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigReloadAutoConfiguration.java:63]: Couldn't we use `@ConditionalOnKubernetesAndConfigEnabled` here too?

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]
Sorry one more thing I just noticed

**Pull request merged**